1) H Philip, T Snir, M Gordin, M Shugay, A Zilberberg, S Efroni

A T-cell repertoire timestamp is at the core of responsiveness to CTLA-4 blockade

iScience (2021)

2) Nili Tickotsky-Moskovitz, Yoram Louzoun, Shirit Dvorkin, Adi Rotkopf, Amir Asher Kuperman, Sol Efroni

CDR3 and V genes show distinct reconstitution patterns in T cell repertoire post-allogeneic bone marrow transplantation

Immunogenetics (2021)

3) Miri Gordin, Hagit Philip, Alona Zilberberg, Moriah Gidoni, Raanan Margalit, Christopher Clouser, Kristofor Adams, Francois Vigneault, Irun R Cohen, Gur Yaari, Sol Efroni

Breast cancer is marked by specific, Public T-cell receptor CDR3 regions shared by mice and humans

PLOS Computational Biology (2021)

4) Anderson DE, Zilberberg A, Philip H, Gordon M, Fluckiger AC, Efroni S, Diaz-Mitoma F

TCR and HLA analysis of patients in a phase I/IIa trial of a therapeutic CMV vaccine against recurrent glioblastoma (GBM)

(Abstract – ESMO 2020, clinical trial with VBI vaccines)

Annals of Oncology (2020)

5) Levy O, Amit G, Vaknin D, Snir T, Efroni S, Castaldi P, Liu YY, Cohen HY, Bashan A

Age-related loss of gene-to-gene transcriptional coordination among single cells

Nature Metabolism (2020) (In Press)

6) Snir T, Efroni S

T cell repertoire sequencing as a cancer’s liquid biopsy–can we decode what the immune system is coding?

Current Opinion in Systems Biology (2020)

7) Ben-Hamo R, Jacob Berger A, Gavert N, Miller M, Pines G, Oren R, Pikarsky E, Benes CH, Neuman T, Zwang Y, Efroni S, Getz G, Straussman R

Predicting and affecting response to cancer therapy based on pathway-level biomarkers

Nature Communications volume 11, Article number: 3296 (2020)

8) Cohen IR, Efroni S, Atlan H

Immune Computation and COVID-19 Mortality: A Rationale for IVIg

Critical Reviews™ in Immunology 2020

9) Cashman R, Zilberberg A, Priel A, Philip H, Varvak A, Jacob A, Shoval I, Efroni S.

A single nucleotide variant of human PARP1 determines response to PARP inhibitors.

NPJ Precis Oncol. 2020 Apr 27;4:10. 

10) Jessel HR, Aharoni L, Efroni S, Bachelet I.

A modeling algorithm for exploring the architecture and construction of bird nests.

Sci Rep. 2019 Oct 

11) Michalovicz, L.T., K.A. Kelly, S. Vashishtha, R. Ben-Hamo, S. Efroni, J.V. Miller, A.R. Locker, K. Sullivan, G. Broderick, D.B. Miller, and J.P. O'Callaghan, Astrocyte‐specific transcriptome analysis using the ALDH 1L1 bac TRAP mouse reveals novel biomarkers of astrogliosis in response to neurotoxicity.

J Neurochem, 2019. 150(4): p. 420-440.

12) Hacohen, A., R. Cohen, S. Efroni, B. Barzel, and I. Bachelet

Digitizable therapeutics for decentralized mitigation of global pandemics.

Sci Rep, 2019. 9(1): p. 14345.

13) Cohen, I.R. and S. Efroni,

The immune system computes the state of the body: Crowd wisdom, machine learning, and immune cell reference repertoires help manage inflammation. 

Front Immunol, 2019. 10: p. 10.

14) Ben-Hamo, R., A. Zilberberg, H. Cohen, K. Bahar-Shany, C. Wachtel, J. Korach, S. Aviel-Ronen, I. Barshack, D. Barash, K. Levanon, and S. Efroni,

Resistance to paclitaxel is associated with a variant of the gene BCL2 in multiple tumor types.

NPJ Precis Oncol, 2019. 3: p. 12.

15) Baranes, K., D. Hibsh, S. Cohen, T. Yamin, S. Efroni, A. Sharoni, and O. Shefi,

Comparing Transcriptome Profiles of Neurons Interfacing Adjacent Cells and Nanopatterned Substrates Reveals Fundamental Neuronal Interactions

Nano Lett, 2019. 19(3): p. 1451-1459.

16) Zhu, Y., Y. Liu, C. Zhang, J. Chu, Y. Wu, Y. Li, J. Liu, Q. Li, S. Li, Q. Shi, L. Jin, J. Zhao, D. Yin, S. Efroni, F. Su, H. Yao, E. Song, and Q. Liu,

Tamoxifen-resistant breast cancer cells are resistant to DNA-damaging chemotherapy because of upregulated BARD1 and BRCA1.

Nat Commun, 2018. 9(1): p. 1595.

17) Priel, A., M. Gordin, H. Philip, A. Zilberberg, and S. Efroni, 

Network representation of T-Cell repertoire–A novel tool to analyze immune response to cancer formation. 

Front Immunol, 2018. 9: p. 2913.

18) Liberman, A., D. Kario, M. Mussel, J. Brill, K. Buetow, S. Efroni, and U. Nevo, 

Cell studio: A platform for interactive, 3D graphical simulation of immunological processes. 

APL Bioeng, 2018. 2(2): p. 026107.

19) Erhart, F., J. Buchroithner, R. Reitermaier, K. Fischhuber, S. Klingenbrunner, I. Sloma, D. Hibsh, R. Kozol, S. Efroni, G. Ricken, A. Wohrer, C. Haberler, J. Hainfellner, G. Krumpl, T. Felzmann, A.M. Dohnal, C. Marosi, and C. Visus, 

Immunological analysis of phase II glioblastoma dendritic cell vaccine (Audencel) trial: immune system characteristics influence outcome and Audencel up-regulates Th1-related immunovariables. 

Acta Neuropathol Commun, 2018. 6(1): p. 135.

20) Volovitz, I., S. Melzer, S. Amar, J. Bocsi, M. Bloch, S. Efroni, Z. Ram, and A. Tarnok,

 Dendritic cells in the context of human tumors: Biology and experimental tools. 

Int Rev Immunol, 2016. 35(2): p. 116-35.

21) Torres, C.M., A. Biran, M.J. Burney, H. Patel, T. Henser-Brownhill, A.S. Cohen, Y. Li, R. Ben-Hamo, E. Nye, B. Spencer-Dene, P. Chakravarty, S. Efroni, N. Matthews, T. Misteli, E. Meshorer, and P. Scaffidi, 

The linker histone H1. 0 generates epigenetic and functional intratumor heterogeneity. 

Science, 2016. 353(6307).

22) Ochayon, D.E., B.M. Baranovski, P. Malkin, R. Schuster, N. Kalay, R. Ben-Hamo, I. Sloma, J. Levinson, J. Brazg, S. Efroni, E.C. Lewis, and U. Nevo, 

Experimental Support for the Ecoimmunity Theory: Distinct Phenotypes of Nonlymphocytic Cells in SCID and Wild-Type Mice. 

Cell Transplant, 2016. 25(8): p. 1575-88.

23) Hibsh, D., K.H. Buetow, G. Yaari, and S. Efroni, 

Quantification of read species behavior within whole genome sequencing of cancer genomes for the stratification and visualization of genomic variation.

 Nucleic Acids Res, 2016. 44(9): p. e81.

24) Gibbs-Bar, L., H. Tempelhof, R. Ben-Hamo, Y. Ely, A. Brandis, R. Hofi, G. Almog, T. Braun, E. Feldmesser, S. Efroni, and K. Yaniv, 

Autotaxin-Lysophosphatidic Acid Axis Acts Downstream of Apoprotein B Lipoproteins in Endothelial Cells. 

Arterioscler Thromb Vasc Biol, 2016. 36(10): p. 2058-67.

25) Elhanati, S., R. Ben-Hamo, Y. Kanfi, A. Varvak, R. Glazz, B. Lerrer, S. Efroni, and H.Y. Cohen, Reciprocal Regulation between SIRT6 and miR-122 Controls Liver Metabolism and Predicts Hepatocarcinoma Prognosis. Cell Rep, 2016. 14(2): p. 234-42.

26) Covacu, R., H. Philip, M. Jaronen, J. Almeida, J.E. Kenison, S. Darko, C.C. Chao, G. Yaari, Y. Louzoun, L. Carmel, D.C. Douek, S. Efroni, and F.J. Quintana, System-wide Analysis of the T Cell Response. Cell Rep, 2016. 14(11): p. 2733-44.

27) Ben-Hamo, R., A. Zilberberg, H. Cohen, and S. Efroni, hsa-miR-9 controls the mobility behavior of glioblastoma cells via regulation of MAPK14 signaling elements. Oncotarget, 2016. 7(17): p. 23170-81. 

28) Shamay-Ramot, A., K. Khermesh, H.T. Porath, M. Barak, Y. Pinto, C. Wachtel, A. Zilberberg, T. Lerer-Goldshtein, S. Efroni, E.Y. Levanon, and L. Appelbaum, Fmrp Interacts with Adar and Regulates RNA Editing, Synaptic Density and Locomotor Activity in Zebrafish. PLoS Genet, 2015. 11(12): p. e1005702.

29) Mazan-Mamczarz, K., R.J. Peroutka, J.J. Steinhardt, M. Gidoni, Y. Zhang, E. Lehrmann, A.L. Landon, B. Dai, S. Houng, P.A. Muniandy, S. Efroni, K.G. Becker, and R.B. Gartenhaus, Distinct inhibitory effects on mTOR signaling by ethanol and INK128 in diffuse large B-cell lymphoma. Cell Commun Signal, 2015. 13: p. 15.

30) Hibsh, D., H. Schori, S. Efroni, and O. Shefi, De novo transcriptome assembly databases for the central nervous system of the medicinal leech. Sci Data, 2015. 2: p. 150015.

31) Bida, O., M. Gidoni, D. Ideses, S. Efroni, and D. Ginsberg, A novel mitosis-associated lncRNA, MA-linc1, is required for cell cycle progression and sensitizes cancer cells to Paclitaxel. Oncotarget, 2015. 6(29): p. 27880-90.

32) Ben-Hamo, R. and S. Efroni, MicroRNA regulation of molecular pathways as a generic mechanism and as a core disease phenotype. Oncotarget, 2015. 6(3): p. 1594-604.

33) Ben-Chetrit, N., D. Chetrit, R. Russell, C. Korner, M. Mancini, A. Abdul-Hai, T. Itkin, S. Carvalho, H. Cohen-Dvashi, W.J. Koestler, K. Shukla, M. Lindzen, M. Kedmi, M. Lauriola, Z. Shulman, H. Barr, D. Seger, D.A. Ferraro, F. Pareja, H. Gil-Henn, T. Lapidot, R. Alon, F. Milanezi, M. Symons, R. Ben-Hamo, S. Efroni, F. Schmitt, S. Wiemann, C. Caldas, M. Ehrlich, and Y. Yarden, Synaptojanin 2 is a druggable mediator of metastasis and the gene is overexpressed and amplified in breast cancer. Sci Signal, 2015. 8(360): p. ra7.

34) Tetievsky, A., M. Assayag, R. Ben-Hamo, S. Efroni, G. Cohen, A. Abbas, and M. Horowitz, Heat acclimation memory: do the kinetics of the deacclimated transcriptome predispose to rapid reacclimation and cytoprotection? J Appl Physiol (1985), 2014. 117(11): p. 1262-77.

35) Hibsh, D., H. Schori, S. Efroni, and O. Shefi, Spatial regulation dominates gene function in the ganglia chain. Bioinformatics, 2014. 30(3): p. 310-6.

36) Cohen, H., R. Ben-Hamo, M. Gidoni, I. Yitzhaki, R. Kozol, A. Zilberberg, and S. Efroni, Shift in GATA3 functions, and GATA3 mutations, control progression and clinical presentation in breast cancer. Breast Cancer Res, 2014. 16(6): p. 464.

37) Cashman, R., H. Cohen, R. Ben-Hamo, A. Zilberberg, and S. Efroni, SENP5 mediates breast cancer invasion via a TGFbetaRI SUMOylation cascade. Oncotarget, 2014. 5(4): p. 1071-82.

38) Ben-Hamo, R., M. Gidoni, and S. Efroni, PhenoNet: identification of key networks associated with disease phenotype. Bioinformatics, 2014. 30(17): p. 2399-405.

39) Mascanfroni, I.D., A. Yeste, S.M. Vieira, E.J. Burns, B. Patel, I. Sloma, Y. Wu, L. Mayo, R. Ben-Hamo, S. Efroni, V.K. Kuchroo, S.C. Robson, and F.J. Quintana, IL-27 acts on DCs to suppress the T cell response and autoimmunity by inducing expression of the immunoregulatory molecule CD39. Nat Immunol, 2013. 14(10): p. 1054-63.

40) Efroni, S., D. Meerzaman, C.F. Schaefer, S. Greenblum, M. Soo-Lyu, Y. Hu, C. Cultraro, E. Meshorer, and K.H. Buetow, Systems analysis utilising pathway interactions identifies sonic hedgehog pathway as a primary biomarker and oncogenic target in hepatocellular carcinoma. IET Syst Biol, 2013. 7(6): p. 243-51.

41) Broderick, G., R. Ben-Hamo, S. Vashishtha, S. Efroni, L. Nathanson, Z. Barnes, M.A. Fletcher, and N. Klimas, Altered immune pathway activity under exercise challenge in Gulf War Illness: an exploratory analysis. Brain Behav Immun, 2013. 28: p. 159-69.

42) Ben-Hamo, R. and S. Efroni, MicroRNA-gene association as a prognostic biomarker in cancer exposes disease mechanisms. PLoS Comput Biol, 2013. 9(11): p. e1003351.

43) Feldstein, O., R. Ben-Hamo, D. Bashari, S. Efroni, and D. Ginsberg, RBM38 is a direct transcriptional target of E2F1 that limits E2F1-induced proliferation. Mol Cancer Res, 2012. 10(9): p. 1169-77.

44) Benichou, J., R. Ben-Hamo, Y. Louzoun, and S. Efroni, Rep-Seq: uncovering the immunological repertoire through next-generation sequencing. Immunology, 2012. 135(3): p. 183-91.

45) Ben-Hamo, R. and S. Efroni, Biomarker robustness reveals the PDGF network as driving disease outcome in ovarian cancer patients in multiple studies. BMC Syst Biol, 2012. 6: p. 3.

46) Vainas, O., D. Harel, I.R. Cohen, and S. Efroni, Reactive animation: from piecemeal experimentation to reactive biological systems. Autoimmunity, 2011. 44(4): p. 271-81.

47) Maman, Y., A. Blancher, J. Benichou, A. Yablonka, S. Efroni, and Y. Louzoun, Immune-induced evolutionary selection focused on a single reading frame in overlapping hepatitis B virus proteins. J Virol, 2011. 85(9): p. 4558-66.

48) Greenblum, S.I., S. Efroni, C.F. Schaefer, and K.H. Buetow, The PathOlogist: an automated tool for pathway-centric analysis. BMC Bioinformatics, 2011. 12: p. 133.

49) Efroni, S., R. Ben-Hamo, M. Edmonson, S. Greenblum, C.F. Schaefer, and K.H. Buetow, Detecting cancer gene networks characterized by recurrent genomic alterations in a population. PLoS One, 2011. 6(1): p. e14437.

50) Ben-Hamo, R. and S. Efroni, Gene expression and network-based analysis reveals a novel role for hsa-miR-9 and drug control over the p38 network in glioblastoma multiforme progression. Genome Med, 2011. 3(11): p. 77.

51) Ben-Hamo, R. and S. Efroni, The whole-organism heavy chain B cell repertoire from Zebrafish self-organizes into distinct network features. BMC Syst Biol, 2011. 5: p. 27.
