#### Rep-Seq analysis ####

1. [immunarch](https://immunarch.com/) - Fast and Seamless Exploration of Single-cell and Bulk T-cell/Antibody Immune Repertoires in R

2. [VDJtools](https://vdjtools-doc.readthedocs.io/en/master/#) - a framework for post-analysis of repertoire sequencing data

3. [scRepertoire](https://github.com/ncborcherding/scRepertoire) - scRepertoire is compatible and integrated with the R packages Trex for deep-learning-based autoencoding of the T cell receptor and Ibex for the B cell receptor.

4. [dandelion](https://github.com/zktuong/dandelion) - a python package for analyzing single cell BCR/TCR data from 10x Genomics 5' solution! It streamlines the pre-processing, leveraging some tools from immcantation suite, and integrates with scanpy/anndata for single-cell BCR/TCR analysis. It also includes a couple of functions for visualization.

5. [scirpy](https://github.com/scverse/scirpy) - Scirpy is a package to analyse T cell receptor (TCR) or B cell receptor (BCR) repertoires from single-cell RNA sequencing (scRNA-seq) data in Python. It seamlessly integrates with scanpy and mudata and provides various modules for data import, analysis and visualization.

#### Alignment ####

6. [TRUST4](https://github.com/liulab-dfci/TRUST4) - TRUST4 is a computational tool to analyze TCR and BCR sequences using unselected RNA sequencing data, profiled from fluid and solid tissues, including tumors

7. [MiXCR](https://github.com/milaboratory/mixcr) - MiXCR is a universal software for fast and accurate analysis of raw T- or B- cell receptor repertoire sequencing data. It works with any kind of sequencing data

8. [stitchr](https://github.com/JamieHeather/stitchr) - Stitch together TCR coding nucleotide sequences from V/J/CDR3 information

#### Clustering ####

9. [ClusTCR](https://svalkiers.github.io/clusTCR/) - a Python interface for rapid clustering of large sets of CDR3 sequences with unknown antigen specificity

10. [CompAIRR](https://github.com/uio-bmi/compairr) - CompAIRR (compairr) is a command line tool to compare two sets of adaptive immune receptor repertoires and compute their overlap. It can also identify which sequences are present in which repertoires. Furthermore, CompAIRR can cluster the sequences in a repertoire set. Sequence comparisons can be exact or approximate. CompAIRR has been shown to be very fast and to have a small memory footprint compared to similar tools, when up to 2 differences are allowed.

11. [GLIPH](https://github.com/immunoengineer/gliph) - GLIPH clusters TCRs that are predicted to bind the same MHC-restricted peptide antigen. When multiple donors have contributed to the clusters, and HLA genotypes for those donors are available, GLIPH additionally can provide predictions of which HLA-allele is presenting the antigen.

12. [GIANA](https://github.com/s175573/GIANA) - GIANA allows computationally-efficient TCR clustering and multi-disease repertoire classification by isometric transformation.

13. [tcrdist3](https://github.com/kmayerb/tcrdist3) - tcrdist3 is a python API-enabled toolkit for analyzing T-cell receptor repertoires

14. [https://github.com/amcdavid/CellaRepertorium](https://github.com/amcdavid/CellaRepertorium) - This package contains methods for clustering, pairing and testing single cell RepSeq data, especially as generated by 10X genomics VDJ solution.

15. [enclone](https://10xgenomics.github.io/enclone/) - Accurate and user-friendly computational tool for clonal grouping to study the adaptive immune system

#### Peptide/Epitope recognition/prediction/specificity

16. [TCRex](https://tcrex.biodatamining.be/) - a web tool for the prediction of TCRâ€“epitope recognition

17. [ImRex](https://github.com/pmoris/ImRex) - TCR-epitope recognition prediction using combined sequence input represention for convolutional neural networks

18. [NetTCR](https://services.healthtech.dtu.dk/services/NetTCR-2.0/) - Sequence-based prediction of peptide-TCR binding

19. [GraphMHC](https://github.com/recognizability/GraphMHC) - neoantigen prediction model applying the graph neural network to molecular structure" A hybrid graph attention network + CNN to predict peptides that bind MHC proteins

20. [TcellMatch](https://github.com/theislab/tcellmatch) - TcellMatch is a collection of models to predict antigen specificity of single T cells based on CDR3 sequences and other single cell modalities, such as RNA counts and surface protein counts.

21. [STAPLER](https://github.com/NKI-AI/STAPLER) - Efficient learning of TCR-peptide specificity prediction from full-length TCR-peptide data


22. [alphafold_finetune](https://github.com/phbradley/alphafold_finetune) - Python code for fine-tuning AlphaFold to perform protein-peptide binding predictions.

#### ML ####

23.  [DeepTCR](https://github.com/sidhomj/DeepTCR) - DeepTCR is a python package that has a collection of unsupervised and supervised deep learning methods to parse TCRSeq data. 

24.  [CoNGA](https://github.com/phbradley/conga) - conga was developed to detect correlation between T cell gene expression profile and TCR sequence in single-cell datasets. We have since added support for gamma delta TCRs and for B cells, too.

25.  [TAPIR](https://vcreate.io/tapir/) - TAPIR is a deep neural network that has been trained to understand the relationships between T-cell receptors (TCRs) and targets across a diverse set of targets and sequence representations.

26.  [DeepAIR](https://github.com/TencentAILabHealthcare/DeepAIR) - DeepAIR: a deep-learning framework for effective integration of sequence and 3D structure to enable adaptive immune receptor analysis

27. [Uni-Fold](https://github.com/dptech-corp/Uni-Fold) - We proudly present Uni-Fold as a thoroughly open-source platform for developing protein models beyond AlphaFold. Uni-Fold introduces the following features:

#### Database ####

27. [IPD-IMGT/HLA](https://www.ebi.ac.uk/ipd/imgt/hla/) - The IPD-IMGT/HLA Database provides a specialist database for sequences of the human major histocompatibility complex (MHC) and includes the official sequences named by the WHO Nomenclature Committee For Factors of the HLA System. The IPD-IMGT/HLA Database was originally part of the international ImMunoGeneTics project (IMGT).

28. [hlabud](https://github.com/slowkow/hlabud) - hlabud provides methods to retrieve sequence alignment data from IMGTHLA and convert the data into convenient R matrices ready for downstream analysis.

29. [TCRdb](http://bioinfo.life.hust.edu.cn/TCRdb/#/) - TCRdb is a comprehensive T-cell receptor (TCR) database, the TCR on the surface of T cells recognize antigen and participate in the activation of T cells in immune response. The diversity and antigen specificity of TCRs are mainly determined by the high hypervariable complementarity-determining region 3 (CDR3). TCRdb contains processed sequences of TCRs CDR3 beta chain of human with different phenotypes. We provide an easy-to-use interface to query and browse the database, which will make TCRdb a very useful resource in T cell related researches.

30. [McPAS-TCR](https://github.com/crazyhottommy/TCR-BCR-seq-analysis?tab=readme-ov-file) - A manually curated catalogue of pathology associated T-cell receptor sequences

31. [VDJdb](https://vdjdb.cdr3.net/) - VDJdb is a curated database of T-cell receptor (TCR) sequences with known antigen specificities. The primary goal of VDJdb is to facilitate access to existing information on T-cell receptor antigen specificities, i.e. the ability to recognize certain epitopes in a certain MHC contexts.